src/applications/vaos/new-appointment/redux/actions.js

```diff
@@ -262,55 +268,131 @@ export function openFacilityPageV2(page, uiSchema, schema) {
 
-      const responses = await Promise.all(
-        parentFacilities.map(parent => {
-          const parentId = parent.id;
-          const siteId = parseFakeFHIRId(
-            getIdOfRootOrganization(parentFacilities, parentId),
+      locations = newAppointment.facilities[typeOfCareId] || null;
+
+      if (parentFacilities?.length && !locations) {
+        try {
+          dispatch({
+            type: FORM_FETCH_CHILD_FACILITIES,
+            isFacilityV2Page: true,
+          });
+
+          const responses = await Promise.all(
+            parentFacilities.map(parent => {
+              const parentId = parent.id;
+              const siteId = parseFakeFHIRId(
+                getIdOfRootOrganization(parentFacilities, parentId),
+              );
+              return getSupportedHealthcareServicesAndLocations({
+                siteId,
+                parentId,
+                typeOfCareId,
+                useVSP,
+              });
+            }),
           );
-          return getSupportedHealthcareServicesAndLocations({
-            siteId,
-            parentId,
-            typeOfCareId,
-            useVSP,
+          locations = responses
+            .reduce(function(arr, row) {
+              return arr.concat(row.locations);
+            }, [])
+            .sort((a, b) => {
+              return a.name < b.name ? -1 : 1;
+            });
+        } catch (err) {
+          dispatch({
+            type: FORM_FETCH_CHILD_FACILITIES_FAILED,
+            isFacilityV2Page: true,
           });
-        }),
-      );
+        }
+      }
+
+      const eligibilityDataNeeded = !!locationId || locations?.length === 1;
+
+      if (eligibilityDataNeeded && !locationId) {
+        locationId = locations[0].id;
+      }
 
-      const facilities = responses
-        .reduce(function(arr, row) {
-          return arr.concat(row.locations);
-        }, [])
-        .sort((a, b) => {
-          return a.name < b.name ? -1 : 1;
+      if (!locations?.length) {
+        parentFacilities.forEach(p => {
+          recordEligibilityFailure(
+            'supported-facilities',
+            typeOfCare.name,
+            parseFakeFHIRId(p.id),
+          );
         });
+      }
+
+      const eligibilityChecks =
+        newAppointment.eligibility[`${locationId}_${typeOfCareId}`] || null;
+
+      if (eligibilityDataNeeded && !eligibilityChecks) {
+        const parentId = getParentOfLocation(parentFacilities, locations[0]).id;
+        const siteId = parseFakeFHIRId(
+          getIdOfRootOrganization(parentFacilities, parentId),
+        );
+        eligibilityData = await getEligibilityData(
+          locations.find(location => location.id === locationId),
+          typeOfCareId,
+          siteId,
+          directSchedulingEnabled,
+          useVSP,
+        );
+
+        recordEligibilityGAEvents(eligibilityData, typeOfCareId, siteId);
+        logEligibilityExplanation(eligibilityData, typeOfCareId, locationId);
+      }
```

Comment:
In the case that the user only has location that supports this type of care, perform eligibility checks immediately so we can determine if the user can continue scheduling.